osc-server/src/main/java/org/osc/core/broker/service/servicefunctionchain/UpdateServiceFunctionChainService.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service.servicefunctionchain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.osc.core.broker.model.entities.appliance.VirtualSystem;
+import org.osc.core.broker.model.entities.virtualization.ServiceFunctionChain;
+import org.osc.core.broker.service.ServiceDispatcher;
+import org.osc.core.broker.service.api.UpdateServiceFunctionChainServiceApi;
+import org.osc.core.broker.service.dto.ServiceFunctionChainDto;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.persistence.ServiceFunctionChainEntityMgr;
+import org.osc.core.broker.service.persistence.VirtualSystemEntityMgr;
+import org.osc.core.broker.service.request.AddOrUpdateServiceFunctionChainRequest;
+import org.osc.core.broker.service.response.BaseJobResponse;
+import org.osc.core.broker.service.validator.RequestValidator;
+import org.osc.core.broker.service.validator.ServiceFunctionChainRequestValidator;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component
+public class UpdateServiceFunctionChainService
+		extends ServiceDispatcher<AddOrUpdateServiceFunctionChainRequest, BaseJobResponse>
+		implements UpdateServiceFunctionChainServiceApi {
+
+	RequestValidator<AddOrUpdateServiceFunctionChainRequest, ServiceFunctionChain> validator;
+	@Reference
+	private ServiceFunctionChainRequestValidator validatorFactory;
+
+	@Override
+	public BaseJobResponse exec(AddOrUpdateServiceFunctionChainRequest request, EntityManager em) throws Exception {
+
+		ServiceFunctionChainDto dto = (ServiceFunctionChainDto) request.getDto();
+		if (this.validator == null) {
+			this.validator = this.validatorFactory.create(em);
+		}
+
+		this.validator.validateAndLoad(request);
+
+		ServiceFunctionChain sfc = ServiceFunctionChainEntityMgr.findById(em, dto.getId());
+
+		// make a Id list of existing Virtual systems in sfc -entity
+		List<Long> sfcVsIdList = new ArrayList<Long>();
```

Comment:
not sure i understand the code here. It looks like we only care about vsid matching.

But what if all the vs ids are matching but the order is different.

Can we just clear out all the VS'es in the SFC, and just use the new list?